crates/chain/src/chain_data.rs

```diff
@@ -30,16 +30,6 @@ impl<A: Clone> ChainPosition<&A> {
     }
 }
 
-impl<A: Anchor> ChainPosition<A> {
-    /// Determines the upper bound of the confirmation height.
-    pub fn confirmation_height_upper_bound(&self) -> Option<u32> {
-        match self {
-            ChainPosition::Confirmed(a) => Some(a.confirmation_height_upper_bound()),
-            ChainPosition::Unconfirmed(_) => None,
-        }
-    }
-}
-
```

Comment:
Why is this removed?

Reply:
Looking at the commit message for a7cbeab734361126957e7f5466a494a35dd8e48e, it seems you have gotten confused. We are not implementing `Anchor` for `ChainPosition` here, we are requiring that `A: Anchor` so that method `confirmation_height_upper_bound` is available.

Reply:
My bad.

Reply:
Np, thanks for the correction!